addons/mail/static/tests/mock_server/mail_mock_server.js

```diff
@@ -584,11 +584,16 @@ async function mail_message_update_content(request) {
 
     const { attachment_ids, body, message_id } = await parseRequestParams(request);
     MailMessage.write([message_id], { body, attachment_ids });
+    if (body === "") {
+        MailMessage.write([message_id], { pinned_at: false });
+    }
+    const message = MailMessage.search_read([["id", "=", message_id]])[0];
```

Comment:
Note that you could destructure `const [message] = MailMessage.search_read()`. This is slightly easier to read ;)